diff --git a/Source/Core/DolphinLibretro/Boot.cpp b/Source/Core/DolphinLibretro/Boot.cpp
index e89f116..dc7f3c5 100644
--- a/Source/Core/DolphinLibretro/Boot.cpp
+++ b/Source/Core/DolphinLibretro/Boot.cpp
@@ -271,6 +271,7 @@ bool retro_load_game(const struct retro_game_info* game)
   UICommon::Init();
   Libretro::Log::Init();
 
+#ifdef HAS_LIBMGBA
   if (Libretro::GBPlayer_active)
   {
     Config::SetBase(Config::MAIN_GBA_ROM_PATHS[Config::GBPLAYER_GBA_INDEX], Libretro::GBPlayer_rom_path);
@@ -285,6 +286,7 @@ bool retro_load_game(const struct retro_game_info* game)
   {
     Config::SetBase(Config::MAIN_GBA_ROM_PATHS[Config::GBPLAYER_GBA_INDEX], std::string{});
   }
+#endif
 
   Discord::SetDiscordPresenceEnabled(false);
   Common::SetEnableAlert(false);
